### PR TITLE
add docker cronjob for disk cleaness/add authentication method in clustering recipe

### PR DESCRIPTION
**Pull request by carol-hsu**
Hi @msfuko 

this script is help for docker's disk management 
I already add it in D/C minions, and confirmed that they would be triggered
```
cron:Sep 10 03:12:01 dcs-qaworker5 run-parts(/etc/cron.daily)[27460]: starting docker-clean.cron
cron:Sep 10 03:12:01 dcs-qaworker5 run-parts(/etc/cron.daily)[27477]: finished docker-clean.cron
```

let's wait for it how it works when the disk is less than 20% :jack_o_lantern: 

Thanks
Carol

### CODE REVIEW

**Comment by carol-hsu** [comment]
Hi @hidetosaito 

I add BA authentication in clustering procedure, which the node with smallest ip number will take charge of enabling the authentication.

Thanks

File: kubernetes/recipes/etcd-clustering.rb [hunk 1]
```diff
@@ -1,9 +1,14 @@
 private_ip = node['opsworks']['instance']['private_ip']
 hostname = node['opsworks']['instance']['hostname']
 members = Array.new
+smallest_ip = 255
 
 node['opsworks']['layers']['etcd']['instances'].each do |inst|
 	members << inst[0]+"=http://"+inst[1][:private_ip]+":7001"
+	last_num = inst[1][:private_ip].split('.')[3]
```
Comment (hidetosaito) [inline kubernetes/recipes/etcd-clustering.rb]: Hi @carol-hsu 

i don't like this code, please reconsider the logic.
i found some problem 

1 . this is assumed instances are the same subnet /24, but in the reality, if there is multi-az, it is possible happened there are same host address such as 
```
instanceA : 10.97.217.93
instanceB : 10.97.216.93
```

2 . if host is not launched yet in OpsWorks, ```node['opsworks']['layers']['etcd']['instances'][n]['private_ip']``` is not assigned yet.

sc
 
```
# opsworks-agent-cli get_json | jq '.opsworks.layers.etcd.instances[] | .private_ip'
"10.97.217.93"
"10.97.217.44"
```

so it is possibly happened newer instance might get a smallest host address (ex: 10.97.217.1) and then perform */root/etcd_enable_ba.sh* even if already executed


if you intend to select an only one instance which can execute */root/etcd_enable_ba.sh*, i'd recommend just select first instance

**Comment by carol-hsu** [comment]
Hi @hidetosaito ,

many thanks for your suggestion!

1. you're right, I'll just get the first IP from layer as the one who enable BA
2. I don't think it would be a problem, since opswork layers will not take no-running instance into considering, so does clustering method, which always need ip for communication.
The clustering recipe is executed manually, and each time etcd daemon will be stopped (previous BA is stopped, too), then group current running instances, starting a new cluster again.
In the above case, no matter what, only etcd1 and etcd2 will group together.

Thanks!

**Comment by carol-hsu** [comment]
Hi @hidetosaito 

I change the logic, I have tested it works. 
please check again :smiley_cat: 

Thanks

**Comment by hidetosaito** [comment]
Hi @carol-hsu 

yup, it's much simpler, thank you so much :+1:

**Pull request merged**